src/v/ssx/thread_worker.h

```diff
@@ -95,88 +126,136 @@ class worker_task final : public task_base {
 };
 
 class thread_worker {
-    static constexpr eventfd_t RUNNABLE_INIT = 0;
-    static constexpr eventfd_t RUNNABLE_READY = 1;
-    static constexpr eventfd_t RUNNABLE_STOP = 2;
-
 public:
-    thread_worker()
-      : _alien{ss::engine().alien()}
+    struct config {
+        // If the thread should be pinned to a the same core as the shard it was
+        // created on.
+        bool pin_to_shard_core = false;
+        // NOTE: pthread names must be less than 12 bytes
+        //
+        // Why 12 bytes? pthread only support names of length 16 and we want to
+        // suffix threads with the core it's associated with via `-xxx`.
+        ss::sstring name = "thread";
+    };
+
+    explicit thread_worker()
+      : _alien(&ss::engine().alien())
+      , _cpu_id(::sched_getcpu())
```

Comment:
did you consider passing in a `cpuset` here? that could allow us more flexibility, for example, arranging for all the alien threads to run on dedicated cores, like here:

https://github.com/ceph/ceph/blob/20398601e76a6ff48b3886d10cffdceb68fcafcb/src/crimson/os/alienstore/alien_store.cc#L105-L119

it'd be a minor change in the future, so i don't think it's necessary right now. the more i think about the more a dedicated core(s) makes a lot of sense. communicating through shared memory is _cheap_ but when we pin the transform to a reactor core we _have_ to context switch and suspend the reactor.

Reply:
Interesting, I'd like to get this in as is - then experiment with a cpuset here instead. I don't know what sort of resource allocation strategy is best here.

Reply:
I'll file a followup task for this.

Reply:
https://github.com/redpanda-data/redpanda/issues/13090